cmd/observer/node_builder/observer_builder.go

```diff
@@ -1071,6 +1493,46 @@ func (builder *ObserverServiceBuilder) enqueueRPCServer() {
 	})
 }
 
+func indexerCollectionHandler(entity flow.Entity, collections storage.Collections,
+	transactions storage.Transactions,
+	logger zerolog.Logger) {
```

Comment:
```suggestion
func indexerCollectionHandler(
	entity flow.Entity,
	collections storage.Collections,
	transactions storage.Transactions,
	logger zerolog.Logger,
) {
```

Reply:
let's keep this in the ingestion engine, but as an exported free method that could be passed. Is this similar enough to the functionality on the ingestion engine to reuse it there as well?